File: codegen/results/v002/gemma4_26b_nothink/20260411_180639/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"io"
	"log"
	"os"
	"strconv"
)

// Person represents the structure for each row in the CSV
type Person struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	// Open the input file
	file, err := os.Open("input/data.csv")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	reader := csv.NewReader(file)

	// Read the header row
	_, err = reader.Read()
	if err != nil {
		log.Fatal(err)
	}

	var people []Person

	// Iterate through the remaining rows
	for {
		record, err := reader.Read()
		if err == io.EOF {
			break
		}
		if err != nil {
			log.Fatal(err)
		}

		// Parse Age (index 1)
		age, err := strconv.Atoi(record[1])
		if err != nil {
			log.Fatal(err)
		}

		// Parse Score (index 3)
		score, err := strconv.ParseFloat(record[3], 64)
		if err != nil {
			log.Fatal(err)
		}

		// Create Person object
		p := Person{
			Name:  record[0],
			Age:   age,
			Email: record[2],
			Score: score,
		}
		people = append(people, p)
	}

	// Encode the slice of people to JSON and write to stdout
	encoder := json.NewEncoder(os.Stdout)
	// Set indentation for pretty-printing (optional, but makes output readable)
	encoder.SetIndent("", "  ")
	
	err = encoder.Encode(people)
	if err != nil {
		log.Fatal(err)
	}
}

File: codegen/results/v002/gemma4_26b_nothink/20260411_180639/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * This script reads a CSV file from 'input/data.csv', parses its content,
 * and outputs a JSON array to stdout.
 */

interface UserData {
  Name: string;
  Age: number;
  Email: string;
  Score: number;
}

function solve() {
  const filePath = path.join(process.cwd(), 'input', 'data.csv');

  try {
    // Check if file exists
    if (!fs.existsSync(filePath)) {
      // If the specific path doesn't exist, we can't proceed.
      // In a real environment, we'd throw an error, but here we exit silently 
      // to satisfy "Do not output anything else" if the input is missing.
      process.exit(1);
    }

    const content = fs.readFileSync(filePath, 'utf-8');
    
    // Split content into lines and remove empty lines
    const lines = content.split(/\r?\n/).filter(line => line.trim() !== '');

    if (lines.length < 2) {
      // Only header or no content
      console.log(JSON.stringify([]));
      return;
    }

    // Extract header to identify columns (though task specifies fixed mapping)
    const headers = lines[0].split(',').map(h => h.trim());
    
    const result: UserData[] = [];

    // Iterate through data rows (skipping header)
    for (let i = 1; i < lines.length; i++) {
      const columns = lines[i].split(',').map(c => c.trim());
      
      // Ensure we have the expected number of columns
      if (columns.length === 4) {
        const row: UserData = {
          Name: columns[0],
          Age: parseInt(columns[1], 10),
          Email: columns[2],
          Score: parseFloat(columns[3])
        };
        result.push(row);
      }
    }

    // Output the final JSON array to stdout
    process.stdout.write(JSON.stringify(result, null, 2) + '\n');

  } catch (error) {
    // If any error occurs, we exit. 
    // The requirement is "Do not output anything else" besides valid JSON.
    process.exit(1);
  }
}

solve();